xbmc/AutoSwitch.cpp

```diff
@@ -18,21 +18,29 @@
 #include "settings/SettingsComponent.h"
 #include "view/ViewState.h"
 
-#define METHOD_BYFOLDERS  0
-#define METHOD_BYFILES   1
-#define METHOD_BYTHUMBPERCENT 2
-#define METHOD_BYFILECOUNT 3
-#define METHOD_BYFOLDERTHUMBS 4
+#include <algorithm>
 
-CAutoSwitch::CAutoSwitch(void) = default;
+namespace
+{
 
-CAutoSwitch::~CAutoSwitch(void) = default;
+enum class Method
+{
+  None,
+  ByFolders,
+  ByFiles,
+  ByThumbPercent,
+  ByFileCount,
+  ByFolderThumbs,
+};
```

Comment:
The naming doesn't follow the code guide: https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#83-enums